addons/survey/models/survey_user_input.py

```diff
@@ -162,6 +165,38 @@ def _compute_attempts_info(self):
                 user_input.attempts_number = attempts_number_result.get('attempts_number', 1)
                 user_input.attempts_count = attempts_number_result.get('attempts_count', 1)
 
+    @api.depends_context('uid')
+    @api.depends('survey_id.survey_type', 'survey_id.restrict_user_ids')
+    def _compute_can_access_survey_user_input(self):
+        can_access_user_inputs = self.filtered_domain(self._get_access_domain())
+        can_access_user_inputs.can_access_survey_user_input = True
+        (self - can_access_user_inputs).can_access_survey_user_input = False
+
+    @api.model
+    def _get_access_domain(self):
```

Comment:
Seems we need some user input specific code to ease sub addons because there are stuff specific to user_inputs that are not completely accessible from the survey. However can't we simplify have code only in sub addons, as anyway they do not seem to share anything ?

Idea
* survey: classic ir.rule [('survey_id.can_access_survey', '=', True)] (aka no python code, like other models)
* hr_recruitment_survey (example): additional ir.rule [('can_access_via_applicant_id', '=', True)] -> will be OR-es with previous rule normally, hence adding some new results based specific 'can_access_via_applicant_id' field where we define the search specific to hr_recruitment_survey

It means a bit of both approaches: keep some base rules people could modify, and have "complex" domains computed in python. Seems it would make the rules a bit clearer for people managing ACLs. What do you think ? Seems like it would generate less python and code redundancy, and have a bit more localized code in sub addons. Wild proposal hein.

Reply:
Hello @tde-banana-odoo :wave: 

Are we to understand that in this new approach, the survey addon must remain as it is in master and that bridges have their own python acl which will be 'Or-ed ' if the user belongs to several groups (e.g.: survey officer and recruitment manager)?